parameter/criterion/include/criterion/client/CriterionInterface.h

```diff
@@ -30,17 +30,59 @@
 #pragma once
 
 #include <string>
+#include <map>
+#include <set>
 
-#include "SelectionCriterionTypeInterface.h"
+namespace core
+{
+namespace selection
+{
+namespace criterion
+{
 
-class ISelectionCriterionInterface
+class CriterionInterface
 {
 public:
-    virtual void setCriterionState(int iState) = 0;
-    virtual int getCriterionState() const = 0;
+    /** Type which represent a criterion state part */
+    using Value = std::string;
```

Comment:
It is needed for creation purpose (creation is made through ParameterMgr)
It cannot be defined in internal Criterion header as the client as no access to it, neither in the ParameterMgr. So I think this is the better place.